.github/workflows/autolabel-pullrequests.yml

```diff
@@ -65,62 +82,74 @@ jobs:
               case /^openwrt\/.*/.test(f.filename):
                 console.log("file changed under OPENWRT component : " + f['filename']);
-                lbls.add("component: openwrt");
+                newCompLbls.add("component: openwrt");
                 break;
               case /^lte\/gateway\/.*/.test(f.filename):
               case /^lte\/protos\/.*/.test(f.filename):
               case /^orc8r\/gateway\/c\/.*/.test(f.filename):
               case /^third_party\/gtp_ovs\/.*/.test(f.filename):
                 console.log("file changed under AGW component : " + f['filename']);
-                lbls.add("component: agw");
+                newCompLbls.add("component: agw");
                 break;
+              case /.*\/cloud\/.*/.test(f.filename):
               case /^\.golangci\.yml/.test(f.filename):
               case /^orc8r\/.*/.test(f.filename):
                 console.log("file changed under ORC8R component : " + f['filename']);
-                lbls.add("component: orc8r");
+                newCompLbls.add("component: orc8r");
                 break;
               case /^xwf\/.*/.test(f.filename):
                 console.log("file changed under XWF component : " + f['filename']);
-                lbls.add("component: xwf");
+                newCompLbls.add("component: xwf");
                 break;
               case /^show-tech\/.*/.test(f.filename):
                 console.log("file changed under SHOW-TECH component : " + f['filename']);
-                lbls.add("component: show-tech");
+                newCompLbls.add("component: show-tech");
                 break;
               case /^docs\/.*/.test(f.filename):
                 console.log("file changed under DOCS: " + f['filename']);
-                lbls.add("component: docs");
+                newCompLbls.add("component: docs");
                 break;
              } // end of switch case
             } // end of for loop
 
-            if( lbls.size == 0 )  {
-              console.log("No components changed in this PR: " + context.issue.number);
+            if( newCompLbls.size == 0 )  {
+              console.log("No new component files changed in this PR: " + context.issue.number);
             }
             else  {
-              let uniqLbls = Array.from(lbls);
+              for (const l of curCompLbls) {
+                if(!newCompLbls.has(l))  {
+                  console.log("Removing the Label:" + l + " since its no more applicable for the PR: " + context.issue.number);
+                  github.issues.removeLabel({
+                    issue_number: context.issue.number,
+                    owner: context.repo.owner,
+                    repo: context.repo.repo,
+                    name: l,
+                  });
+                } // end of if block
+              } // end of for loop
 
-              github.issues.setLabels({
+              let uniqLbls = Array.from(newCompLbls);
+              github.issues.addLabels({
```

Comment:
Are you sure your code will not duplicate labels that were already present before?

Reply:
yes @quentinDERORY, because the `newCompLbls` is a Set too. Why you think it could duplicate ?

Reply:
because you are using add api, and if there are labels already present on the PR, it might get duplicated? no?
I would recommend only adding the ones that are not already present on the PR.

Reply:
you are right @quentinDERORY. I fixed it now to avoid duplicate labels due to the add. Please check now.